[
   {
      "body" : "Thanks for the issue! I saw this failure earlier at least twice (only on travis, though)",
      "created_at" : "2016-09-29T18:51:53Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8842#issuecomment-250557677",
      "id" : 250557677,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8842",
      "updated_at" : "2016-09-29T18:51:53Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/250557677",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/6399679?v=3",
         "events_url" : "https://api.github.com/users/MarcoFalke/events{/privacy}",
         "followers_url" : "https://api.github.com/users/MarcoFalke/followers",
         "following_url" : "https://api.github.com/users/MarcoFalke/following{/other_user}",
         "gists_url" : "https://api.github.com/users/MarcoFalke/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/MarcoFalke",
         "id" : 6399679,
         "login" : "MarcoFalke",
         "organizations_url" : "https://api.github.com/users/MarcoFalke/orgs",
         "received_events_url" : "https://api.github.com/users/MarcoFalke/received_events",
         "repos_url" : "https://api.github.com/users/MarcoFalke/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/MarcoFalke/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/MarcoFalke/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/MarcoFalke"
      }
   },
   {
      "body" : "Oops, I'll take a look...",
      "created_at" : "2016-09-29T19:21:39Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8842#issuecomment-250565306",
      "id" : 250565306,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8842",
      "updated_at" : "2016-09-29T19:21:39Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/250565306",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/7463573?v=3",
         "events_url" : "https://api.github.com/users/sdaftuar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/sdaftuar/followers",
         "following_url" : "https://api.github.com/users/sdaftuar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/sdaftuar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/sdaftuar",
         "id" : 7463573,
         "login" : "sdaftuar",
         "organizations_url" : "https://api.github.com/users/sdaftuar/orgs",
         "received_events_url" : "https://api.github.com/users/sdaftuar/received_events",
         "repos_url" : "https://api.github.com/users/sdaftuar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/sdaftuar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/sdaftuar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/sdaftuar"
      }
   },
   {
      "body" : "Going to reopen this based on https://travis-ci.org/bitcoin/bitcoin/jobs/164463351",
      "created_at" : "2016-10-02T20:40:11Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8842#issuecomment-250994398",
      "id" : 250994398,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8842",
      "updated_at" : "2016-10-02T20:40:11Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/250994398",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/6399679?v=3",
         "events_url" : "https://api.github.com/users/MarcoFalke/events{/privacy}",
         "followers_url" : "https://api.github.com/users/MarcoFalke/followers",
         "following_url" : "https://api.github.com/users/MarcoFalke/following{/other_user}",
         "gists_url" : "https://api.github.com/users/MarcoFalke/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/MarcoFalke",
         "id" : 6399679,
         "login" : "MarcoFalke",
         "organizations_url" : "https://api.github.com/users/MarcoFalke/orgs",
         "received_events_url" : "https://api.github.com/users/MarcoFalke/received_events",
         "repos_url" : "https://api.github.com/users/MarcoFalke/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/MarcoFalke/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/MarcoFalke/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/MarcoFalke"
      }
   },
   {
      "body" : "Ugh, I will take another look, thanks for noticing.",
      "created_at" : "2016-10-03T14:55:11Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8842#issuecomment-251128738",
      "id" : 251128738,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8842",
      "updated_at" : "2016-10-03T14:55:11Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/251128738",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/7463573?v=3",
         "events_url" : "https://api.github.com/users/sdaftuar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/sdaftuar/followers",
         "following_url" : "https://api.github.com/users/sdaftuar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/sdaftuar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/sdaftuar",
         "id" : 7463573,
         "login" : "sdaftuar",
         "organizations_url" : "https://api.github.com/users/sdaftuar/orgs",
         "received_events_url" : "https://api.github.com/users/sdaftuar/received_events",
         "repos_url" : "https://api.github.com/users/sdaftuar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/sdaftuar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/sdaftuar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/sdaftuar"
      }
   },
   {
      "body" : "Ok I think I see what I did wrong in #8854 -- syncing with a ping doesn't ensure that all block announcements have made it to us, just that any that have been announced have been processed.\r\n\r\nI'm working on a new fix (my idea is to replace that sync_with_ping() with logic to wait until the tip has been announced), but I'm having trouble reproducing the problem.",
      "created_at" : "2016-10-03T20:00:06Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8842#issuecomment-251209963",
      "id" : 251209963,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8842",
      "updated_at" : "2016-10-03T20:00:06Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/251209963",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/7463573?v=3",
         "events_url" : "https://api.github.com/users/sdaftuar/events{/privacy}",
         "followers_url" : "https://api.github.com/users/sdaftuar/followers",
         "following_url" : "https://api.github.com/users/sdaftuar/following{/other_user}",
         "gists_url" : "https://api.github.com/users/sdaftuar/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/sdaftuar",
         "id" : 7463573,
         "login" : "sdaftuar",
         "organizations_url" : "https://api.github.com/users/sdaftuar/orgs",
         "received_events_url" : "https://api.github.com/users/sdaftuar/received_events",
         "repos_url" : "https://api.github.com/users/sdaftuar/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/sdaftuar/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/sdaftuar/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/sdaftuar"
      }
   },
   {
      "body" : "Please let me know if this is still an issue.",
      "created_at" : "2016-10-11T08:52:09Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8842#issuecomment-252851126",
      "id" : 252851126,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8842",
      "updated_at" : "2016-10-11T08:52:09Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/252851126",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/6399679?v=3",
         "events_url" : "https://api.github.com/users/MarcoFalke/events{/privacy}",
         "followers_url" : "https://api.github.com/users/MarcoFalke/followers",
         "following_url" : "https://api.github.com/users/MarcoFalke/following{/other_user}",
         "gists_url" : "https://api.github.com/users/MarcoFalke/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/MarcoFalke",
         "id" : 6399679,
         "login" : "MarcoFalke",
         "organizations_url" : "https://api.github.com/users/MarcoFalke/orgs",
         "received_events_url" : "https://api.github.com/users/MarcoFalke/received_events",
         "repos_url" : "https://api.github.com/users/MarcoFalke/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/MarcoFalke/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/MarcoFalke/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/MarcoFalke"
      }
   },
   {
      "body" : "@sdaftuar - p2p-compactblocks just failed in a travis build for me: https://travis-ci.org/bitcoin/bitcoin/builds/170131004 . The next build (which was just a squash of the same commits) succeeded: https://travis-ci.org/bitcoin/bitcoin/builds/170143716 , so it appears there's still an intermittent problem with the compact blocks testcase. Sadly there doesn't appear to be much logging in travis and I can't reproduce this locally.",
      "created_at" : "2016-10-24T14:38:30Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8842#issuecomment-255759241",
      "id" : 255759241,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8842",
      "updated_at" : "2016-10-24T14:38:30Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/255759241",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/1063656?v=3",
         "events_url" : "https://api.github.com/users/jnewbery/events{/privacy}",
         "followers_url" : "https://api.github.com/users/jnewbery/followers",
         "following_url" : "https://api.github.com/users/jnewbery/following{/other_user}",
         "gists_url" : "https://api.github.com/users/jnewbery/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/jnewbery",
         "id" : 1063656,
         "login" : "jnewbery",
         "organizations_url" : "https://api.github.com/users/jnewbery/orgs",
         "received_events_url" : "https://api.github.com/users/jnewbery/received_events",
         "repos_url" : "https://api.github.com/users/jnewbery/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/jnewbery/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/jnewbery/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/jnewbery"
      }
   },
   {
      "body" : "Those travis builds were from a PR from this branch: https://github.com/HackerResidency/bitcoin/blob/add-p2sh-segwit-options-to-bitcoin-tx/qa/rpc-tests/p2p-compactblocks.py , which includes https://github.com/bitcoin/bitcoin/pull/8882 .",
      "created_at" : "2016-10-24T14:42:12Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8842#issuecomment-255760306",
      "id" : 255760306,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8842",
      "updated_at" : "2016-10-24T14:42:12Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/255760306",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/1063656?v=3",
         "events_url" : "https://api.github.com/users/jnewbery/events{/privacy}",
         "followers_url" : "https://api.github.com/users/jnewbery/followers",
         "following_url" : "https://api.github.com/users/jnewbery/following{/other_user}",
         "gists_url" : "https://api.github.com/users/jnewbery/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/jnewbery",
         "id" : 1063656,
         "login" : "jnewbery",
         "organizations_url" : "https://api.github.com/users/jnewbery/orgs",
         "received_events_url" : "https://api.github.com/users/jnewbery/received_events",
         "repos_url" : "https://api.github.com/users/jnewbery/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/jnewbery/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/jnewbery/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/jnewbery"
      }
   },
   {
      "body" : "Looked into the failure, but so far couldn't find a problem in the test setup that would cause it.\r\n\r\nThe [failure](https://travis-ci.org/bitcoin/bitcoin/jobs/170131009#L1501) happened in [line 138](https://github.com/bitcoin/bitcoin/blob/d39162bb7ac11ef745e203e3607f0ff9f1a551c8/qa/rpc-tests/test_framework/util.py#L138) of the `sync_blocks` util function when the two test nodes failed to sync after 60 seconds. The place where the failing `sync_blocks` was called was [line 809](https://github.com/bitcoin/bitcoin/blob/d39162bb7ac11ef745e203e3607f0ff9f1a551c8/qa/rpc-tests/p2p-compactblocks.py#L809) of run_test after segwit was activated on node 1. Node 1 should have added more blocks than node 0 at that point, so some kind of failure must have happened to cause node 0 not to receive (or accept) the new blocks from node 1, and the sync_block function to time out.\r\n\r\nSince I don't have any good theories on what would cause node 0 not to sync with node 1, I'm currently running the test in a loop to try to reproduce the problem. \r\n\r\nI added some debug prints and made the following change to avoid a different (unrelated) intermittent test failure:\r\n\r\n```\r\n--- a/qa/rpc-tests/p2p-compactblocks.py\r\n+++ b/qa/rpc-tests/p2p-compactblocks.py\r\n@@ -300,8 +300,8 @@ class CompactBlocksTest(BitcoinTestFramework):\r\n             assert(segwit_tx_generated) # check that our test is not broken\r\n \r\n         # Wait until we've seen the block announcement for the resulting tip\r\n-        tip = int(self.nodes[0].getbestblockhash(), 16)\r\n-        assert(self.test_node.wait_for_block_announcement(tip))\r\n+        tip = int(node.getbestblockhash(), 16)\r\n+        assert(test_node.wait_for_block_announcement(tip))\r\n \r\n         # Now mine a block, and look at the resulting compact block.\r\n         test_node.clear_block_announcement()\r\n```",
      "created_at" : "2016-10-24T20:41:53Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8842#issuecomment-255859919",
      "id" : 255859919,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8842",
      "updated_at" : "2016-10-24T20:41:53Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/255859919",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/7133040?v=3",
         "events_url" : "https://api.github.com/users/ryanofsky/events{/privacy}",
         "followers_url" : "https://api.github.com/users/ryanofsky/followers",
         "following_url" : "https://api.github.com/users/ryanofsky/following{/other_user}",
         "gists_url" : "https://api.github.com/users/ryanofsky/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/ryanofsky",
         "id" : 7133040,
         "login" : "ryanofsky",
         "organizations_url" : "https://api.github.com/users/ryanofsky/orgs",
         "received_events_url" : "https://api.github.com/users/ryanofsky/received_events",
         "repos_url" : "https://api.github.com/users/ryanofsky/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/ryanofsky/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/ryanofsky/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/ryanofsky"
      }
   },
   {
      "body" : "Update: running the test overnight on my workstation showed the `sync_blocks` call returning true after 3 loop iterations in 1060 out of 1089 runs, and returning true after 2 loop iterations in the other 29 runs. It never got anywhere close to the 60 iterations that would have been required to cause the timeout.\r\n\r\n\r\n\r\n",
      "created_at" : "2016-10-25T17:36:00Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8842#issuecomment-256105282",
      "id" : 256105282,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8842",
      "updated_at" : "2016-10-25T17:36:00Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/256105282",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/7133040?v=3",
         "events_url" : "https://api.github.com/users/ryanofsky/events{/privacy}",
         "followers_url" : "https://api.github.com/users/ryanofsky/followers",
         "following_url" : "https://api.github.com/users/ryanofsky/following{/other_user}",
         "gists_url" : "https://api.github.com/users/ryanofsky/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/ryanofsky",
         "id" : 7133040,
         "login" : "ryanofsky",
         "organizations_url" : "https://api.github.com/users/ryanofsky/orgs",
         "received_events_url" : "https://api.github.com/users/ryanofsky/received_events",
         "repos_url" : "https://api.github.com/users/ryanofsky/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/ryanofsky/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/ryanofsky/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/ryanofsky"
      }
   },
   {
      "body" : "This issue should be closed now that #9058 is merged.",
      "created_at" : "2016-11-11T15:27:19Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8842#issuecomment-259981448",
      "id" : 259981448,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8842",
      "updated_at" : "2016-11-11T15:27:19Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/259981448",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/7133040?v=3",
         "events_url" : "https://api.github.com/users/ryanofsky/events{/privacy}",
         "followers_url" : "https://api.github.com/users/ryanofsky/followers",
         "following_url" : "https://api.github.com/users/ryanofsky/following{/other_user}",
         "gists_url" : "https://api.github.com/users/ryanofsky/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/ryanofsky",
         "id" : 7133040,
         "login" : "ryanofsky",
         "organizations_url" : "https://api.github.com/users/ryanofsky/orgs",
         "received_events_url" : "https://api.github.com/users/ryanofsky/received_events",
         "repos_url" : "https://api.github.com/users/ryanofsky/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/ryanofsky/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/ryanofsky/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/ryanofsky"
      }
   },
   {
      "body" : "Yes, let's be optimistic and close it :)",
      "created_at" : "2016-11-11T16:35:10Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/issues/8842#issuecomment-259997273",
      "id" : 259997273,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/8842",
      "updated_at" : "2016-11-11T16:35:10Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/259997273",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/126646?v=3",
         "events_url" : "https://api.github.com/users/laanwj/events{/privacy}",
         "followers_url" : "https://api.github.com/users/laanwj/followers",
         "following_url" : "https://api.github.com/users/laanwj/following{/other_user}",
         "gists_url" : "https://api.github.com/users/laanwj/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/laanwj",
         "id" : 126646,
         "login" : "laanwj",
         "organizations_url" : "https://api.github.com/users/laanwj/orgs",
         "received_events_url" : "https://api.github.com/users/laanwj/received_events",
         "repos_url" : "https://api.github.com/users/laanwj/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/laanwj/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/laanwj/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/laanwj"
      }
   }
]
